### PR TITLE
Fix array slicing with some slice values (especially negative bounds)

### PR DESCRIPTION
**Pull request by pitrou**
For example `array[:-1]` would return an empty array.

### CODE REVIEW

**Comment by seibert** [comment]
This PR also needs a remerge with master now.

File: numba/tests/test_indexing.py [hunk 1]
```diff
@@ -148,18 +140,23 @@ def test_1d_slicing(self, flags=enable_pyobj_flags):
...
+                        (9, 0, -1),
+                        (-5, -2, 1),
+                        (0, -1, 1),
+                        ]:
+            expected = pyfunc(a, *indices)
+            self.assertPreciseEqual(cfunc(a, *indices), expected)
 
     def test_1d_slicing_npm(self):
         """
         Return of arbitrary array is not supported yet
```
Comment (seibert) [inline numba/tests/test_indexing.py]: I don't think this docstring applies anymore.

**Comment by seibert** [comment]
Aside from my question about the docstring, I think this is ready to merge.

**Comment by pitrou** [comment]
Oops, sorry, I had forgotten about this.

**Pull request merged**